cpp/Makefile

```diff
@@ -0,0 +1,10 @@
+CFLAGS=`gdal-config --cflags`
```

Comment:
For production, we want to compile with optimizations turned on `-O3`.

Also, it would be convenient to have warnings enabled, ie. `-Wall`